src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/PSUserAgent.cs

```diff
@@ -131,7 +132,8 @@ internal static string PlatformName
                     // only generate the windows user agent once
                     if(s_windowsUserAgent == null){
                         // find the version in the windows operating system description
-                        string versionText = OS.Substring(OS.LastIndexOf(" ") +1);
+                        Regex pattern = new Regex(@"\d+(\.\d+)+");
+                        string versionText = pattern.Match(OS).Value;
                         Version windowsPlatformversion = new Version(versionText);
```

Comment:
Can we trust OSDescription? Maybe add Assert at least?

Reply:
I believe we can for all the versions of Windows this will run on. What assert did you have in mind?

Reply:
Diagnostics.Assert(versionText != null, "...")

Reply:
Hmm it would probably be better to wrap it in a try/catch because if the system was detected as Windows, then returning `Windows NT` without the version is better than nothing at all.